### PR TITLE
Implement Trezor backend for hardware-based staking and create a skeleton for a device CLI

### PR DESCRIPTION
**Pull request by tuxxy**
### What this does:

1. Creates a `TrustedDevice` ABC for integrating trusted hardware devices (e.g., trezor, yubihsm, etc).
2. Implements a `Trezor` backend with signing, verification, and device init functions.
3. Adds a `nucypher device` CLI entry point skeleton
4. Adds tests for all the above.
5. Adds a dependency for `trezor[ethereum]`.

### CODE REVIEW

File: nucypher/cli/hardware/backends.py [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+
+from abc import ABC, abstractmethod
+from collections import namedtuple
+from functools import wraps
+from importlib import import_module
+
+from nucypher.crypto.signing import InvalidSignature
+
+
+class TrustedDevice(ABC):
```
Comment (KPrasch) [inline nucypher/cli/hardware/backends.py]: :boom:

File: nucypher/cli/hardware/backends.py [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+            self.trezor_tools = import_module('trezorlib.tools')
+
+            self.client = self.trezor_client.get_default_client()
+            self.__bip44_path = self.trezor_tools.parse_path(self.DEFAULT_BIP44_PATH)
+        except TransportException:
+            raise RuntimeError("Could not find a TREZOR device to connect to. Have you unlocked it?")
+        except ModuleNotFoundError:
+            raise RuntimeError("The nucypher package wasn't installed with the 'trezor' extra.")
+
+    def _handle_device_call(device_func):
```
Comment (KPrasch) [inline nucypher/cli/hardware/backends.py]: :+1: - Nice

File: tests/cli/device/test_trezor_backend.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+from nucypher.cli.hardware.backends import Trezor
+from nucypher.crypto.signing import InvalidSignature
+
+
+def test_trezor_defaults(mock_trezorlib):
+    trezor_backend = Trezor()
+
+    assert trezor_backend.DEFAULT_BIP44_PATH == "m/44'/60'/0'/0"
+    assert trezor_backend._Trezor__bip44_path == [2147483692, 2147483708,
+                                                  2147483648, 0]
```
Comment (cygnusv) [inline tests/cli/device/test_trezor_backend.py]: What do these numbers mean?

Comment (tuxxy) [reply tests/cli/device/test_trezor_backend.py]: It's the derived BIP44 path in numerical form.

File: nucypher/cli/hardware/backends.py [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+    NuCypher network.
+
+    TODO: Define an abstractmethod for BIP44 derivation?
+    """
+
+    # We intentionally keep the address index off the path so that the
+    # subclass interfaces can handle which address index to use.
+    DEFAULT_BIP44_PATH = "m/44'/60'/0'/0"
+
+    Signature = namedtuple('Signature', ['signature', 'address'])
```
Comment (KPrasch) [inline nucypher/cli/hardware/backends.py]: You know I love named tuples...

File: tests/cli/conftest.py [hunk 5]
```diff
@@ -38,6 +43,23 @@
...
+    class mock_usb1:
+
+        class USBErrorNoDevice(Exception):
+            pass
+
+        class USBErrorBusy(Exception):
+            pass
+
+    usb1 = mock_usb1()
+    sys.modules['usb1'] = usb1
```
Comment (KPrasch) [inline tests/cli/conftest.py]: Hahah... Mocks like this always make me giggle.

Perhaps this can be moved to a test module outside of CLI?

**Pull request merged**